pkg/machine/ssh.go

```diff
@@ -27,9 +27,15 @@ func CommonSSH(username, identityPath, name string, sshPort int, inputArgs []str
 	cmd := exec.Command("ssh", args...)
 	logrus.Debugf("Executing: ssh %v\n", args)
 
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	cmd.Stdin = os.Stdin
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
 
-	return cmd.Run()
+	err := cmd.Run()
+	if err != nil {
+		return fmt.Errorf("SSH failed: %w (stdout: %q, stderr: %q)", err, stdout.String(), stderr.String())
+	}
+
+	return nil
```

Comment:
You cannot do this, this breaks machine ssh. The input must be stdin and output must be streamed to stdout/err otherwise no user will ever get the output and can use interactively.
And in the error case we already print stdout/err to stderr/out so I don't see how this adds any more context other than `SSH failed`.

Reply:
Ouch, didn't see that `machine ssh` is using that. Thanks for catching!